This PR does not seem to be sufficient to make this file pass PHPStan Level 9:

```console
$ vendor/bin/phpstan analyse --level 9 ./lib/favicons.php 
Note: Using configuration file /home/alex/GitHub/FreshRSS/phpstan.neon.
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ------------------------------------------------------------------------------------------------ 
  Line   favicons.php                                                                                    
 ------ ------------------------------------------------------------------------------------------------ 
  :16    Parameter #1 $finfo of function finfo_buffer expects finfo, finfo|false given.                  
  :16    Parameter #1 $haystack of function strpos expects string, string|false given.                   
  :17    Parameter #1 $finfo of function finfo_close expects finfo, finfo|false given.                   
  :32    Parameter #1 $handle of function curl_setopt_array expects CurlHandle, CurlHandle|false given.  
  :40    Parameter #1 $handle of function curl_setopt_array expects CurlHandle, CurlHandle|false given.  
  :41    Parameter #1 $handle of function curl_exec expects CurlHandle, CurlHandle|false given.          
  :42    Parameter #1 $handle of function curl_getinfo expects CurlHandle, CurlHandle|false given.       
  :43    Parameter #1 $handle of function curl_close expects CurlHandle, CurlHandle|false given.         
  :50    Function downloadHttp() should return string but returns bool|string.                           
 ------ ------------------------------------------------------------------------------------------------ 


                                                                                                                        
 [ERROR] Found 9 errors
```

--- a/lib/favicons.php
+++ b/lib/favicons.php
@@ -12,8 +12,11 @@ function isImgMime(string $content): bool {
 	}
 	$isImage = true;
 	try {
+		/** @var finfo $fInfo */
 		$fInfo = finfo_open(FILEINFO_MIME_TYPE);
-		$isImage = strpos(finfo_buffer($fInfo, $content), 'image') !== false;
+		/** @var string $content */
+		$content = finfo_buffer($fInfo, $content);
+		$isImage = strpos($content, 'image') !== false;
 		finfo_close($fInfo);
 	} catch (Exception $e) {
 		echo 'Caught exception: ',  $e->getMessage(), "\n";
@@ -28,6 +31,7 @@ function downloadHttp(string &$url, array $curlOptions = []): string {
 	if (!$url) {
 		return '';
 	}
+	/** @var CurlHandle $ch */
 	$ch = curl_init($url);
 	curl_setopt_array($ch, [
 			CURLOPT_RETURNTRANSFER => true,
@@ -38,6 +42,7 @@ function downloadHttp(string &$url, array $curlOptions = []): string {
 			CURLOPT_ENCODING => '',	//Enable all encodings
 		]);
 	curl_setopt_array($ch, $curlOptions);
+	/** @var string $response */
 	$response = curl_exec($ch);
 	$info = curl_getinfo($ch);
 	curl_close($ch);

--- a/p/f.php
+++ b/p/f.php
@@ -32,6 +32,10 @@ if ($ico_mtime == false || $ico_mtime < $txt_mtime || ($ico_mtime < time() - (mt
 
 	// no ico file or we should download a new one.
 	$url = file_get_contents($txt);
+	if ($url === false) {
+		show_default_favicon(1800);
+		exit();
+	}
 	if (!download_favicon($url, $ico)) {
 		// Download failed
 		if ($ico_mtime == false) {

